Sources/performance-worker/WorkerService.swift

```diff
@@ -98,3 +123,52 @@ final class WorkerService: Grpc_Testing_WorkerService.ServiceProtocol, Sendable
     throw RPCError(code: .unimplemented, message: "This RPC has not been implemented yet.")
   }
 }
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+extension WorkerService {
+  private func serverSetup(_ config: Grpc_Testing_ServerConfig) async throws {
+    try self.state.withLockedValue { state in
+      switch state.role {
+      case .server(_):
+        throw RPCError(code: .alreadyExists, message: "A server has been previously set up.")
+      case .client(_):
+        throw RPCError(code: .failedPrecondition, message: "This worker has a client set up.")
+      default:
+        ()
+      }
+    }
+    // The asynchronous function 'run()' can't be called inside the synchronous
+    // closure passed to 'withLockedValue'.
+    let server = GRPCServer(transports: [], services: [BenchmarkService()])
+    try await server.run()
```

Comment:
This isn't quite right because this will suspend until the server has finished running, which means that in `runServer` we won't continue reading the request messages.

Reply:
Should I add a Task group in 'runServer' that has a Task that calls runServer()?